datacube/scripts/spindex.py

```diff
@@ -0,0 +1,190 @@
+    help="Populate the spatial index after creation (slow). "
+         "For finer grained updating, use the 'spindex update' command"
+)
+@click.argument('srids', type=int, nargs=-1)
+@ui.pass_index()
+def create(index: Index, update: bool, srids: Sequence[int]):
+    if not index.supports_spatial_indexes:
+        echo("The active index driver does not support spatial indexes")
+        exit(1)
+    if not srids:
+        echo("Must supply at least one CRS to create/update")
+        exit(1)
+
+    confirmed = []
+    failed = []
+    for srid in srids:
+        try:
+            crs = CRS(f"epsg:{srid}")
+        except pyproj.exceptions.CRSError:
+            failed.append(srid)
+            continue
+        if crs in index.spatial_indexes():
+            # A spatial index for crs already exists: skip silently
+            confirmed.append(crs)
+        elif index.create_spatial_index(crs):
+            # Creation succeeded
+            confirmed.append(crs)
+        else:
+            # Creation attempted but failed
+            failed.append(srid)
+    if failed:
+        str_failed = ','.join(f'epsg:{srid}' for srid in failed)
+        echo(f"Could not create spatial indexes for: {str_failed}")
+    if confirmed:
+        echo(f"Spatial indexes created for: {','.join(str(crs.epsg) for crs in confirmed)}")
+    if update and failed:
+        echo("Skipping update")
+    elif update:
+        result = index.update_spatial_index(confirmed)
+        echo(f'{result} extents checked and updated in spatial indexes')
+    else:
+        echo("Newly created spatial indexes are unpopulated - run 'datacube spindex update' before use.")
+    exit(len(failed))
+
+
+@system.command(
+    "list",
+    help="List all CRSs for which spatial indexes exist in this index"
+)
+@ui.pass_index()
+def list_spindex(index):
+    if not index.supports_spatial_indexes:
+        echo("The active index driver does not support spatial indexes")
+        exit(1)
+    for crs in index.spatial_indexes():
+        echo(f'epsg:{crs.epsg}')
+    exit(0)
+
+
+@system.command(
+    'update',
+    help='Update a spatial index for particular CRSs '
+         '(express CRSs as EPSG numbers - e.g. 3857, not "epsg:3857").')
+@click.option(
+    '--product', '-p', multiple=True,
+    help="The name of a product to update the spatial index for (can be used multiple times for multiple products"
+)
+@click.option(
+    '--dataset', '-d', multiple=True,
+    help="The id of a dataset to update the spatial index for (can be used multiple times for multiple datasets"
+)
+@click.argument('srids', type=int, nargs=-1)
+@ui.pass_index()
+def update(index: Index, product: Sequence[str], dataset: Sequence[str], srids: Sequence[int]):
+    if not index.supports_spatial_indexes:
+        echo("The active index driver does not support spatial indexes")
+        exit(1)
+    if not srids:
+        echo("Must supply at least one CRS to create/update")
```

Comment:
This message implies that a spatial index will be created if it doesn't exist, but the logic doesn't support this

Reply:
Yep.